src/crimson/osd/pg_recovery.cc

```diff
@@ -154,6 +154,10 @@ size_t PGRecovery::start_primary_recovery_ops(
 	out->push_back(recover_missing(soid, item.need));
 	++started;
       }
+    } else {
```

Comment:
@xxhdx1985126 how about transpose these two branches and flatten these conditions? so the code reads
```c++
  if (pg->get_recovery_backend()->is_recovering(soid)) {
    auto& recovery_waiter = pg->get_recovery_backend()->get_recovering(soid);
    out->push_back(recovery_waiter.wait_for_recovered_blocking());
    ++started;
  } else if (pg->get_recovery_backend()->is_recovering(head)) {
    ++skipped;
  } else {
    out->push_back(recover_missing(soid, item.need));
    ++started;
  }
 
```

Reply:
@tchaikov Done:-)